src/poetry/repositories/link_sources/base.py

```diff
@@ -39,21 +40,16 @@ class LinkSource:
 
     def __init__(self, url: str) -> None:
         self._url = url
+        self._get_link_cache_wrapper = functools.lru_cache(maxsize=1)(
+            self._get_link_cache
+        )
 
     @property
     def url(self) -> str:
         return self._url
 
     def versions(self, name: str) -> Iterator[Version]:
```

Comment:
I reckon this takes a `NormalizedName` already so can be annotated as such, saving the need to re-canonicalize.